src/huggingface_hub/repository.py

```diff
@@ -226,6 +226,27 @@ def is_git_ignored(filename: Union[str, Path]) -> bool:
     return is_ignored
 
 
+def is_binary_file(filename: Union[str, Path]) -> bool:
```

Comment:
Other than reading the first few bytes, I think this answer gives a more accurate way of checking if the file is binary or not: https://stackoverflow.com/a/7392391/2536294

Reply:
I implemented what you propose in 9fe6bb6, let me know if that's what you had in mind. Cc @coyotte508